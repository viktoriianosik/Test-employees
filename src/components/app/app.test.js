import React from "react";
import renderer from "react-test-renderer";
import App from "./app";
import * as reactRedux from "react-redux";
import testUsers from "../../test-data/test-data";

describe(`Render component App correctly`, () => {
  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
  const mockedDispatch = jest.fn();
  
  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear()
  });

  it(`Should App connected to store render correctly`, () => {
    useSelectorMock.mockReturnValue({
      users: testUsers,
      selectedUsers: testUsers,
    });
    useDispatchMock.mockReturnValue(mockedDispatch);
    const tree = renderer.create(
      <App />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});