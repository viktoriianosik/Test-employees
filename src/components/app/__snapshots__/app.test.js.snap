// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component App correctly Should App connected to store render correctly 1`] = `
Array [
  <div
    className="employee"
  >
    <h2
      className="employee__title title"
    >
      Employees
    </h2>
    <ul
      className="users-list"
    >
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          A
        </h3>
        <label
          className="user-item__name"
        >
          Asi Bob
          <input
            checked={true}
            className="user-item__input visually-hidden"
            onChange={[Function]}
            type="checkbox"
            value="222"
          />
          <span
            className="user-item__checkbox"
          />
        </label>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          B
        </h3>
        <label
          className="user-item__name"
        >
          Bort Alex
          <input
            checked={true}
            className="user-item__input visually-hidden"
            onChange={[Function]}
            type="checkbox"
            value="111"
          />
          <span
            className="user-item__checkbox"
          />
        </label>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          C
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          D
        </h3>
        <label
          className="user-item__name"
        >
          Dark Anna
          <input
            checked={true}
            className="user-item__input visually-hidden"
            onChange={[Function]}
            type="checkbox"
            value="444"
          />
          <span
            className="user-item__checkbox"
          />
        </label>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          E
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          F
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          G
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          H
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          I
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          J
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          K
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          L
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          M
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          N
        </h3>
        <label
          className="user-item__name"
        >
          Nofl Mark
          <input
            checked={true}
            className="user-item__input visually-hidden"
            onChange={[Function]}
            type="checkbox"
            value="333"
          />
          <span
            className="user-item__checkbox"
          />
        </label>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          O
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          P
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          Q
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          R
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          S
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          T
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          U
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          V
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          W
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          X
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          Y
        </h3>
        <p>
          -----
        </p>
      </li>
      <li
        className="users-item"
      >
        <h3
          className="users__letter"
        >
          Z
        </h3>
        <p>
          -----
        </p>
      </li>
    </ul>
  </div>,
  <div
    className="employees-birthday"
  >
    <h2
      className="employees-birthday__title title"
    >
      Employees birthday
    </h2>
    <h3>
      February
    </h3>
    <ul
      className="birthday-list"
    >
      <li
        className="birthday-item"
      >
        Bort Alex - 23 February, 2010 year
      </li>
    </ul>
    <h3>
      March
    </h3>
    <ul
      className="birthday-list"
    >
      <li
        className="birthday-item"
      >
        Nofl Mark - 29 March, 2014 year
      </li>
    </ul>
    <h3>
      June
    </h3>
    <ul
      className="birthday-list"
    >
      <li
        className="birthday-item"
      >
        Asi Bob - 17 June, 2020 year
      </li>
    </ul>
    <h3>
      September
    </h3>
    <ul
      className="birthday-list"
    >
      <li
        className="birthday-item"
      >
        Dark Anna - 10 September, 2018 year
      </li>
    </ul>
  </div>,
]
`;
